model_api/cpp/models/include/models/classification_model.h

```diff
@@ -31,6 +32,34 @@ struct ClassificationResult;
 struct ResultBase;
 struct ImageInputData;
 
+struct HierarchicalConfig  {
+    std::map<std::string, int> label_to_idx;
+    std::vector<std::pair<std::string, std::string>> label_tree_edges;
+    std::vector<std::vector<std::string>> all_groups;
+    std::map<int, std::pair<int,int>> head_idx_to_logits_range;
+    int num_multiclass_heads;
+    int num_multilabel_heads;
+    int num_single_label_classes;
```

Comment:
Can you move from ints to size_t? I hope this will make json validate that the values are positive.

Reply:
Done